### PR TITLE
fix(warehouse): fix upload error and add additional logs

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Additional logs for upload error

## Notion Ticket

https://www.notion.so/rudderstacks/Additional-logs-for-warehouse-upload-errors-18daf9877eb04009b9e7fb431e72c02e?pvs=4

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/upload.go [hunk 1]
```diff
@@ -1615,7 +1630,7 @@ func (job *UploadJobT) setUploadError(statusError error, state string) (string,
 
 	// Reset the state as aborted if max retries
 	// exceeded.
-	uploadErrorAttempts := 5
+	uploadErrorAttempts := uploadErrors[state]["attempt"].(int)
```
Comment (achettyiitr) [inline warehouse/upload.go:1633]: This got pushed by mistake https://github.com/rudderlabs/rudder-server/pull/2956

File: warehouse/upload.go [hunk 2]
```diff
@@ -1716,7 +1731,6 @@ func (job *UploadJobT) setUploadError(statusError error, state string) (string,
 
 	// On aborted state, validate credentials to allow
 	// us to differentiate between user caused abort vs platform issue.
-	state = model.Aborted
```
Comment (achettyiitr) [inline warehouse/upload.go:1719]: This got pushed by mistake https://github.com/rudderlabs/rudder-server/pull/2956

File: warehouse/upload.go [hunk 3]
```diff
@@ -1595,7 +1594,23 @@ func (job *UploadJobT) Aborted(attempts int, startTime time.Time) bool {
 }
 
 func (job *UploadJobT) setUploadError(statusError error, state string) (string, error) {
-	pkgLogger.Errorf("[WH]: Failed during %s stage: %v\n", state, statusError.Error())
+	defer func() {
+		pkgLogger.Errorw("upload error",
+			logfield.JobID, job.upload.ID,
```
Comment (lvrach) [inline warehouse/upload.go:1599]: Should this be a warning? Given the impact and the frequency of upload errors.

I am cornered that jobID can be confused with the job_id of sources, should we name it upload_job_id to make it unambiguous? 

```suggestion
		pkgLogger.Errorw("upload error",
			logfield.UploadJobID, job.upload.ID,
```

File: warehouse/upload.go [hunk 4]
```diff
@@ -1595,7 +1594,23 @@ func (job *UploadJobT) Aborted(attempts int, startTime time.Time) bool {
...
+			logfield.Status, state,
```
Comment (lvrach) [inline warehouse/upload.go:1600]: prefixing with `upload_` to remove ambiguity: 
```suggestion
			logfield.UploadStatus, state,
```

**Review by lvrach — APPROVED** [review]

**Review by abhimanyubabbar — APPROVED** [review]

**Pull request merged**